cmd/soroban-rpc/internal/methods/util.go

```diff
@@ -0,0 +1,42 @@
+package methods
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/db"
+)
+
+func getProtocolVersion(
+	ctx context.Context,
+	ledgerEntryReader db.LedgerEntryReader,
+	ledgerReader db.LedgerReader,
+) (uint32, error) {
+	readTx, err := ledgerEntryReader.NewCachedTx(ctx)
+	if err != nil {
+		return 0, err
+	}
+	defer func() {
+		doneErr := readTx.Done()
+		err = errors.Join(err, doneErr)
+	}()
+
+	latestLedger, err := readTx.GetLatestLedgerSequence()
+	if err != nil {
+		return 0, err
+	}
+
+	// obtain bucket size
+	closeMeta, ok, err := ledgerReader.GetLedger(ctx, latestLedger)
+	if err != nil {
+		return 0, err
+	}
+	if !ok {
+		return 0, fmt.Errorf("missing meta for latest ledger (%d)", latestLedger)
+	}
+	if closeMeta.V != 1 {
+		return 0, fmt.Errorf("latest ledger (%d) meta has unexpected verion (%d)", latestLedger, closeMeta.V)
+	}
+	return uint32(closeMeta.V1.LedgerHeader.Header.LedgerVersion), nil
```

Comment:
This routine seems really heavy just to get the latest protocol version :/ Can we cache it? Also it shares a lot of code with `getBucketListSizeAndProtocolVersion`; at the very least this can use that after the `closeMeta` acquisition, yeah?

Reply:
Oh I just saw that you're using a cached tx. That seems fine then? If you could throw in a benchmark that'd be sick

Reply:
It uses a cached transaction which will cache the entries when requested multiple times, but not the first time.

Reply:
@Shaptic  Did some quick benchmarking for `getProtocolVersion`. Not bad at all as it is fetching data from cached tx.

```

goos: darwin
goarch: arm64
pkg: github.com/stellar/soroban-rpc/cmd/soroban-rpc/internal/methods
BenchmarkGetProtocolVersion
BenchmarkGetProtocolVersion-12    	   58597	     19727 ns/op
```

Reply:
Added benchmark in `util_test.go`